pkg/actuators/machine/machine_scope.go

```diff
@@ -204,7 +208,46 @@ func (s *machineScope) getUserData() ([]byte, error) {
 		return nil, fmt.Errorf("The userData secret %s missing %s key", objKey, userDataSecretKey)
 	}
 
-	return userData, nil
+	ignData, err := s.getIgnitionData(userData)
+	if err != nil {
+		klog.Errorf("Failed to generate ignition user data. error: %w", err)
+		return nil, err
```

Comment:
Return the wrapped error. Do not only log it and return the parent error only.

Reply:
changed